### PR TITLE
Add standard github CI/CD

### PR DESCRIPTION
**Pull request by bennettgoble**
Wire up basic Github tooling.

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  pull_request:
+  push:
+    branches: [main]
+    tags: ["v*.*.*"]
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        os: [windows-2022, macos-12, ubuntu-22.04]
```
[the remark this answers is not in the document]
Comment (callumlinden) [reply .github/workflows/build.yaml:13]: 13 is required for CEF (mostly due to the Xcode requirement) and 14 was released September 26th although I haven't upgraded yet.

[the remark this answers is not in the document]
Comment (bennettgoble) [reply .github/workflows/build.yaml]: Alright, then it probably makes sense to start moving any new builds to macos-13. I've bumped the version here. Thanks, Callum.

[the remark this answers is not in the document]
Comment (nat-goodspeed) [reply .github/workflows/build.yaml]: Is macOS runner version something that dependabot can surface in our consuming repos?

[the remark this answers is not in the document]
Comment (bennettgoble) [reply .github/workflows/build.yaml]: Nope, not yet anyways.

**Pull request merged**